x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/grouping/Categorize.java

```diff
@@ -156,4 +176,175 @@ public Expression field() {
     public String toString() {
         return "Categorize{field=" + field + "}";
     }
+
+    public GroupingKey.Supplier groupingKey(Function<Expression, ExpressionEvaluator.Factory> toEvaluator) {
+        return mode -> new GroupingKeyFactory(source(), toEvaluator.apply(field), mode);
+    }
+
+    record GroupingKeyFactory(Source source, ExpressionEvaluator.Factory field, AggregatorMode mode) implements GroupingKey.Factory {
+        @Override
+        public GroupingKey apply(DriverContext context, int resultOffset) {
+            ExpressionEvaluator field = this.field.get(context);
+            CategorizeEvaluator evaluator = null;
+            TokenListCategorizer.CloseableTokenListCategorizer categorizer = null;
+            try {
+                categorizer = new TokenListCategorizer.CloseableTokenListCategorizer(
+                    new CategorizationBytesRefHash(new BytesRefHash(2048, context.bigArrays())),
+                    CategorizationPartOfSpeechDictionary.getInstance(),
+                    0.70f
+                );
+                evaluator = new CategorizeEvaluator(
```

Comment:
It seems the CategorizeEvaluator will be executed twice: here and in `toEvaluator`?

Reply:
I was wondering the same.

However, when running the `CategorizeOperatorTests` it seems that `Categorize::toEvaluator` is never executed.